application/controllers/File_manager.php

```diff
@@ -0,0 +1,25 @@
+<?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
+class File_manager extends CI_Controller
```

Comment:
Not sure if this was tested but usually CI forces class names for controllers to be without special characters e.g "_"  in its current state this controller returns `Unable to locate the specified class: File_manager.php`

Reply:
In fact this file was not intended to be uploaded 😢 . In my plan, the project was devided into 2 parts. One parts for the model & driver, another for view & template. The pathset was for the model & driver.

Reply:
From Jan. 31 to Feb. 6 is a traditional vacation here for me. In this period of time I may finish the most key part of the project, afterwards, I may finish the rest of the functions in late Feb. or Mar.
Would you allow to merge them as two patch? I may prefer to delete the controller in this patch, and complete it in the next one, or I may push the end to end code together later.

Reply:
The unfinished controller has been removed.